source/tdx-guest-hardening.rst

```diff
@@ -1000,4 +1000,325 @@ https://github.com/intel/ccc-linux-guest-hardening/tree/master/bkc/kafl/userspac
+	KAFL_CTL=/sys/kernel/debug/kafl
+	echo “VSOCK fuzzing harness” | hcat
+	echo "start"  > $KAFL_CTL/control
+	socat - VSOCK-CONNECT:2:8089
+	echo "done"  > $KAFL_CTL/control
+
+Now it should be possible to start up a new VSOCK harness by first,
+start listening on the host using  `socat VSOCK-LISTEN:8089,fork –`,
+and then start kAFL (make sure it’s using HARNESS_NONE, as always when
+using userspace harnesses) using `fuzz.sh run linux-guest --debug -p1
+--sharedir sharedir/`. You should see the text “VSOCK fuzzing harness”
+appear in your kAFL process.
+
+To summarize these steps can be executed to start a VSOCK harness:
+
+On the guest:
+
+.. code-block:: bash
+
+	socat VSOCK-LISTEN:8089,fork –
+
+On the host:
+
+.. code-block:: bash
+
+	socat - vsock-accept:3:8089
+
+On the guest:
+
+.. code-block:: bash
+
+	socat - VSOCK-CONNECT:2:8089
+
+On the host:
+
+.. code-block:: bash
+
+	socat VSOCK-LISTEN:8089,fork –
+
+It is also likely that in the above-mentioned setup the kAFL fuzzer
+will not make any progress. This is due to the fact that the inputs
+are not stable. This happens due to the fact that an external process
+is part of the fuzzing setup. If you encounter this issue, you might
+need to modify kAFL slightly. In the function `execute()` in
+`$BKC_ROOT/kafl/fuzzer/kafl_fuzzer/worker/worker.py`, when a value is
+assigned to the variable `stable`, make sure to overwrite this with
+True. It is also possible to add a custom command line flag enabling
+this feature to the kAFL settings in
+`$BKC_ROOT/kafl/fuzzer/kafl_fuzzer/common/config.py`.
+
+The above example for the virtio-vsock has demonstrated how to enable
+fuzzing in a more complex driver setup scenario using a userspace kAFL
+harness. The end output of the fuzzing step is a set of reproducible
+crashes that a fuzzer finds for the
+given driver. The crashes needs to be investigated and the ones that
+are determined
+to be real security issues need to be fixed in the code.
+
+4. **Perform code fixes**. Based on the above steps 2 and 3, a set of
+hardening patches
+need to be created to fix the identified issues. We strongly encourage to submit
+any such hardening patches to the mainline Linux kernel to ensure
+everyone benefits
+the joint hardened kernel, as well to get suggestions on the most
+appropriate way of
+fixing these issues. Also in order to verify that the issues have been
+addressed by
+respective patches, a new round of fuzzing needs to be performed to
+verify that the
+issues found in step 3 are not reproducible anymore.
+
+5. **Enable driver in the TDX filter**. When the driver code has been
+hardened and all
+the patches are integrated and verified, the driver can be enabled in
+the TDX guest by modifying
+the allow list in the TDX driver filter code in arch/x86/kernel/tdx-filter.c.
+
+**Example**. For the virtio-vsock driver the following patch adds it
+to the list of
+allowed devices on the virtio bus.
+
+   .. code-block:: c
```

Comment:
This results in
~~~
tdx-guest-hardening.rst:1289: WARNING: Could not lex literal_block as "c". Highlighting skipped.
~~~

I propose to use `diff` as the langage here.